### DISCUSSION TITLE
LDAP on Samba 4 (AD)

### DISCUSSION THREAD

**Discussion by compumatter** [Leantime/leantime#969] [Q&A]
Hello,

We love the looks of this product but LDAP is a requirement.  We can't get it going with Samba 4 which has divorced itself from using OpenLdap and now uses AD.

Is there a clear cut example of what AD configurations look like in the configuration.php file ?

**Comment by marcelfolaron** [comment]
Where does is break down? The documentation is here https://docs.leantime.io/#/installation/configuration?id=ldap-configuration-beta

AD does work, we have a few folks in our discord channel who were able to set it up. I haven't tested it in Samba. 

Do you get any error messages? Anything in resources/logs/error.log?

**Reply by compumatter** [reply]
I have an experienced LDAP developer looking at the code right now.  I will see what he comes up with and report back.  For now I was getting PHP Warning:  ldap_bind(): Unable to bind to server: Can't contact LDAP server in ...src/domain/ldap/services/class.ldap.php on line 119

Ldap config looks like this:
``
    public $useLdap = true;
    public $ldapType = 'AD'; //Select the correct directory type. Currently Supported: OL - OpenLdap, AD - Active Directory
    public $ldapHost = 'anacortes.office.mydomain.net'; //FQDN
    public $ldapPort = 636; //Default Port
    public $baseDn = 'dc=office,dc=mydomain,dc=net'; //Base DN, example: DC=example,DC=com
    public $ldapDn = 'cn=users,dc=office,dc=mydomain,dc=net'; //Location of users, example: CN=users,DC=example,DC=com
    public $ldapUserDomain = 'office.mydomain.net'; //Domain after ldap, example @example.com
    public $bindUser = 'Administrator@office.mydomain.net'; //ldap user that can search directory. (Should be read only)
    public $bindPassword = 'myBind#Password';

    //Default ldap keys in your directory.
    public $ldapKeys = '{
        "username":"sAMAccountName",
        "groups":"memberof",
        "email":"mail",
        "firstname":"givenName",
        "lastname":"sn",
        "phonenumber":""
        }';

    //Default role assignments upon first login. (Optional) Can be updated in user settings for each user
    public $ldapLtGroupAssignments = '{
          "5": {
            "ltRole":"readonly",
            "ldapRole":""
          },
          "10": {
            "ltRole":"commenter",
            "ldapRole":""
          },
          "20": {
            "ltRole":"editor",
            "ldapRole":""
          },
          "30": {
            "ltRole":"manager",
            "ldapRole":""
          },
          "40": {
            "ltRole":"admin",
            "ldapRole":"proj-m-a"
          },
          "50": {
            "ltRole":"owner",
            "ldapRole":"administrators"
          }
        }';
``

**Comment by marcelfolaron** [comment]
One thing I notice is that you are missing part of your domain in the dn
definitions.
Since you have 4 parts in your domain you’d need to add dc=anacortes to
baseDN and to ldapdn as well.

**Comment by compumatter** [comment]
Hello,

We did add that 4th addition per your thoughts.  Unfortunately so far same error on ...class.ldap.php on line 119

We integrate 3 other programs with [Samba AD LDAP](https://wiki.samba.org/index.php/Setting_up_Samba_as_an_Active_Directory_Domain_Controller) on this same environment including NextCloud, Resourcespace and Mediawiki.  

Is there any other LDAP testing or debug features within the software that may reveal a more specific reason for failure in the existing setup?

My dev so far has not got it working.
